Added support for IPv6.

madscientistninja opened the pull request:
This pull request enables IPv6 to be used. If your using a IPv6 address without a domain name it must be formatted like this gopher://[::1]/, if you use gopher://::1/ it will result in a error.
#2

madscientistninja committed:
Add's IPv6 support.

IPv6 connections needs to use AF_INET6 instead of AF_INET so a function called "is_it_ipv6" was added to test the host to see if IPv4 or IPv6 was used and then uses a AF_INET6 instead if IPv6 is detected in fetch_data. Also entry2url needed to be modified as a IPv6 address like gopher://[::1]/ would turn into gopher://::1/ which is not accessible.

madscientistninja committed:
Updated readme to include the IPv6 feature.

felixplesoianu commented:
Hi! I tested your additions, and in Python 3 I get the following error:

>   File "gophersnake.py", line 128
>     if (is_it_ipv6(e[3], port, True)==True):
>                                            ^
> TabError: inconsistent use of tabs and spaces in indentation

Sorry about that; I know it's unusual, but Gophersnake uses tabs for indentation. Could you please fix and resubmit? (Unless there's some way I can do it myself.) Python 3 compatibility is essential; if anything, I'd be willing to drop Python 2.

There are some issues of coding style too, but I can fix those myself. Was going to brush up the code a little anyway.

madscientistninja committed:
Fixed indentation and python 3 compatibilty.

Fixed inconsistent use of tabs and spaces in indentation and python 3 compatibility.

madscientistninja commented:
I made a new commit that should fix the indentation problem.
